### DISCUSSION TITLE
cdk app monorepo with CD

### DISCUSSION THREAD

**Discussion by ahammond** [aws/aws-pdk#790] [Q&A]
We have over 100 cdk app repos deploying uncounted thousands of stacks. Each cdk app repo has it's own dedicated CDK pipeline. I'd like to start consolidating some of these cdk app repos into a cdk app monorepo. I need to find a way to do this which doesn't involve re-writing the cdk pipelines. I would like a better answer around how the underlying CodePipelines are triggered (ideally, if one of the apps is not "affected", then it's pipeline should not trigger).

Also related and of interest to is is the identification of orphaned stacks for culling.

**Comment by agdimech** [comment]
Hi,

I am not aware of any way you could achieve this without some conditional logic being added to your pipeline (source triggers do not support conditional execution as far as I am aware). I would suspect it would look like a build action that is triggered from your monorepo source changing and it perform the following:

- `nx affected` command to see whether the target project is in the list of affected projects.
- If affected, continue the workflow to build and deploy your code otherwise skip pipeline.

The only thing with this is that if you make a change to Project A, then it will trigger a pipeline execution for your other 99 project pipelines (however they should all be skipped).

Sorry I can't be of more help.
